In [1]:
import numpy as np
import pandas as pd

In [5]:
test = ['FirstCA', 'SecondCA', 'ThirdCA']
numbers_of_student = np.arange(1,41)
np.random.seed(1)
score = np.random.randint(1,10,120)
score = score.reshape(40,3)

student_score = pd.DataFrame(data = score, index= numbers_of_student, columns = test)
student_score

FirstCA  SecondCA  ThirdCA
1         6         9        6
2         1         1        2
3         8         7        3
4         5         6        3
5         5         3        5
6         8         8        2
7         8         1        7
8         8         7        2
9         1         2        9
10        9         4        9
11        8         4        7
12        6         2        4
13        5         9        2
14        5         1        4
15        3         1        5
16        3         8        8
17        9         7        4
18        8         8        5
19        6         4        7
20        9         1        3
21        8         8        8
22        4         1        9
23        8         8        2
24        2         4        1
25        9         7        5
26        6         7        3
27        6         8        9
28        5         5        8
29        8         5        1
30        3         1        8
31        2         8        9
32        5         1        2
33        9         3        4
34        2         3        8
35        3         7        1
36        3         7        7
37        3         8        8
38        1         7        6
39        2         5        7
40        1         7        6

In [6]:
student_score['FirstCA']#RETRIEVING A COLUMN

1     6
2     1
3     8
4     5
5     5
6     8
7     8
8     8
9     1
10    9
11    8
12    6
13    5
14    5
15    3
16    3
17    9
18    8
19    6
20    9
21    8
22    4
23    8
24    2
25    9
26    6
27    6
28    5
29    8
30    3
31    2
32    5
33    9
34    2
35    3
36    3
37    3
38    1
39    2
40    1
Name: FirstCA, dtype: int32

In [8]:
student_score[['FirstCA']]#RETRIEVING A COLUMN IN 2D

FirstCA
1         6
2         1
3         8
4         5
5         5
6         8
7         8
8         8
9         1
10        9
11        8
12        6
13        5
14        5
15        3
16        3
17        9
18        8
19        6
20        9
21        8
22        4
23        8
24        2
25        9
26        6
27        6
28        5
29        8
30        3
31        2
32        5
33        9
34        2
35        3
36        3
37        3
38        1
39        2
40        1

In [9]:
student_score[['FirstCA', 'ThirdCA']]#RETRIEVING MORE THAN A COLUMN

FirstCA  ThirdCA
1         6        6
2         1        2
3         8        3
4         5        3
5         5        5
6         8        2
7         8        7
8         8        2
9         1        9
10        9        9
11        8        7
12        6        4
13        5        2
14        5        4
15        3        5
16        3        8
17        9        4
18        8        5
19        6        7
20        9        3
21        8        8
22        4        9
23        8        2
24        2        1
25        9        5
26        6        3
27        6        9
28        5        8
29        8        1
30        3        8
31        2        9
32        5        2
33        9        4
34        2        8
35        3        1
36        3        7
37        3        8
38        1        6
39        2        7
40        1        6

In [11]:
student_score.loc[7] #retrieving a row from a set index

FirstCA     8
SecondCA    1
ThirdCA     7
Name: 7, dtype: int32

In [12]:
student_score.iloc[7] #retrieving a row from a default index

FirstCA     8
SecondCA    7
ThirdCA     2
Name: 8, dtype: int32

In [13]:
student_score.loc[[7,15,18,20,25,28,35]]

FirstCA  SecondCA  ThirdCA
7         8         1        7
15        3         1        5
18        8         8        5
20        9         1        3
25        9         7        5
28        5         5        8
35        3         7        1

In [14]:
student_score.iloc[9:15,0:2] #retrieving a subset of rows and columns

FirstCA  SecondCA
10        9         4
11        8         4
12        6         2
13        5         9
14        5         1
15        3         1

In [16]:
student_score['Exam'] = np.random.randint(30,70,40).reshape(40,1)
student_score

FirstCA  SecondCA  ThirdCA  Exam
1         6         9        6    43
2         1         1        2    63
3         8         7        3    50
4         5         6        3    40
5         5         3        5    62
6         8         8        2    55
7         8         1        7    51
8         8         7        2    69
9         1         2        9    62
10        9         4        9    49
11        8         4        7    55
12        6         2        4    44
13        5         9        2    43
14        5         1        4    50
15        3         1        5    36
16        3         8        8    43
17        9         7        4    45
18        8         8        5    54
19        6         4        7    39
20        9         1        3    32
21        8         8        8    37
22        4         1        9    35
23        8         8        2    66
24        2         4        1    67
25        9         7        5    51
26        6         7        3    43
27        6         8        9    57
28        5         5        8    43
29        8         5        1    44
30        3         1        8    53
31        2         8        9    46
32        5         1        2    58
33        9         3        4    33
34        2         3        8    32
35        3         7        1    44
36        3         7        7    30
37        3         8        8    59
38        1         7        6    41
39        2         5        7    65
40        1         7        6    59

In [17]:
student_score['Total_score'] = student_score['FirstCA'] + student_score['SecondCA'] + student_score['ThirdCA'] + student_score['Exam']

student_score

FirstCA  SecondCA  ThirdCA  Exam  Total_score
1         6         9        6    43           64
2         1         1        2    63           67
3         8         7        3    50           68
4         5         6        3    40           54
5         5         3        5    62           75
6         8         8        2    55           73
7         8         1        7    51           67
8         8         7        2    69           86
9         1         2        9    62           74
10        9         4        9    49           71
11        8         4        7    55           74
12        6         2        4    44           56
13        5         9        2    43           59
14        5         1        4    50           60
15        3         1        5    36           45
16        3         8        8    43           62
17        9         7        4    45           65
18        8         8        5    54           75
19        6         4        7    39           56
20        9         1        3    32           45
21        8         8        8    37           61
22        4         1        9    35           49
23        8         8        2    66           84
24        2         4        1    67           74
25        9         7        5    51           72
26        6         7        3    43           59
27        6         8        9    57           80
28        5         5        8    43           61
29        8         5        1    44           58
30        3         1        8    53           65
31        2         8        9    46           65
32        5         1        2    58           66
33        9         3        4    33           49
34        2         3        8    32           45
35        3         7        1    44           55
36        3         7        7    30           47
37        3         8        8    59           78
38        1         7        6    41           55
39        2         5        7    65           79
40        1         7        6    59           73

In [18]:
np.max(student_score['Total_score'])

86

In [19]:
age = np.random.randint(12,15,40)
index = 0
student_score.insert(loc=index, column= 'Age', value = age)

student_score

Age  FirstCA  SecondCA  ThirdCA  Exam  Total_score
1    13        6         9        6    43           64
2    14        1         1        2    63           67
3    12        8         7        3    50           68
4    12        5         6        3    40           54
5    14        5         3        5    62           75
6    12        8         8        2    55           73
7    13        8         1        7    51           67
8    12        8         7        2    69           86
9    14        1         2        9    62           74
10   12        9         4        9    49           71
11   14        8         4        7    55           74
12   14        6         2        4    44           56
13   12        5         9        2    43           59
14   13        5         1        4    50           60
15   12        3         1        5    36           45
16   12        3         8        8    43           62
17   12        9         7        4    45           65
18   13        8         8        5    54           75
19   14        6         4        7    39           56
20   13        9         1        3    32           45
21   12        8         8        8    37           61
22   12        4         1        9    35           49
23   12        8         8        2    66           84
24   13        2         4        1    67           74
25   14        9         7        5    51           72
26   12        6         7        3    43           59
27   12        6         8        9    57           80
28   14        5         5        8    43           61
29   14        8         5        1    44           58
30   14        3         1        8    53           65
31   14        2         8        9    46           65
32   13        5         1        2    58           66
33   14        9         3        4    33           49
34   12        2         3        8    32           45
35   14        3         7        1    44           55
36   12        3         7        7    30           47
37   12        3         8        8    59           78
38   12        1         7        6    41           55
39   12        2         5        7    65           79
40   13        1         7        6    59           73

In [21]:
student_score.loc['AVG'] = (np.mean(student_score['Age']), np.mean(student_score['FirstCA']), np.mean(student_score['SecondCA']), np.mean(student_score['ThirdCA']), np.mean(student_score['Exam']), np.mean(student_score['Total_score']))

student_score

Age  FirstCA  SecondCA  ThirdCA  Exam  Total_score
1    13.00    6.000     9.000    6.000  43.0       64.000
2    14.00    1.000     1.000    2.000  63.0       67.000
3    12.00    8.000     7.000    3.000  50.0       68.000
4    12.00    5.000     6.000    3.000  40.0       54.000
5    14.00    5.000     3.000    5.000  62.0       75.000
6    12.00    8.000     8.000    2.000  55.0       73.000
7    13.00    8.000     1.000    7.000  51.0       67.000
8    12.00    8.000     7.000    2.000  69.0       86.000
9    14.00    1.000     2.000    9.000  62.0       74.000
10   12.00    9.000     4.000    9.000  49.0       71.000
11   14.00    8.000     4.000    7.000  55.0       74.000
12   14.00    6.000     2.000    4.000  44.0       56.000
13   12.00    5.000     9.000    2.000  43.0       59.000
14   13.00    5.000     1.000    4.000  50.0       60.000
15   12.00    3.000     1.000    5.000  36.0       45.000
16   12.00    3.000     8.000    8.000  43.0       62.000
17   12.00    9.000     7.000    4.000  45.0       65.000
18   13.00    8.000     8.000    5.000  54.0       75.000
19   14.00    6.000     4.000    7.000  39.0       56.000
20   13.00    9.000     1.000    3.000  32.0       45.000
21   12.00    8.000     8.000    8.000  37.0       61.000
22   12.00    4.000     1.000    9.000  35.0       49.000
23   12.00    8.000     8.000    2.000  66.0       84.000
24   13.00    2.000     4.000    1.000  67.0       74.000
25   14.00    9.000     7.000    5.000  51.0       72.000
26   12.00    6.000     7.000    3.000  43.0       59.000
27   12.00    6.000     8.000    9.000  57.0       80.000
28   14.00    5.000     5.000    8.000  43.0       61.000
29   14.00    8.000     5.000    1.000  44.0       58.000
30   14.00    3.000     1.000    8.000  53.0       65.000
31   14.00    2.000     8.000    9.000  46.0       65.000
32   13.00    5.000     1.000    2.000  58.0       66.000
33   14.00    9.000     3.000    4.000  33.0       49.000
34   12.00    2.000     3.000    8.000  32.0       45.000
35   14.00    3.000     7.000    1.000  44.0       55.000
36   12.00    3.000     7.000    7.000  30.0       47.000
37   12.00    3.000     8.000    8.000  59.0       78.000
38   12.00    1.000     7.000    6.000  41.0       55.000
39   12.00    2.000     5.000    7.000  65.0       79.000
40   13.00    1.000     7.000    6.000  59.0       73.000
AVG  12.85    5.275     5.075    5.225  48.7       64.275

In [22]:
student_score.drop('AVG',axis = 0,inplace=True)
student_score.drop('Total_score',axis = 1,inplace=True)

student_score

Age  FirstCA  SecondCA  ThirdCA  Exam
1   13.0      6.0       9.0      6.0  43.0
2   14.0      1.0       1.0      2.0  63.0
3   12.0      8.0       7.0      3.0  50.0
4   12.0      5.0       6.0      3.0  40.0
5   14.0      5.0       3.0      5.0  62.0
6   12.0      8.0       8.0      2.0  55.0
7   13.0      8.0       1.0      7.0  51.0
8   12.0      8.0       7.0      2.0  69.0
9   14.0      1.0       2.0      9.0  62.0
10  12.0      9.0       4.0      9.0  49.0
11  14.0      8.0       4.0      7.0  55.0
12  14.0      6.0       2.0      4.0  44.0
13  12.0      5.0       9.0      2.0  43.0
14  13.0      5.0       1.0      4.0  50.0
15  12.0      3.0       1.0      5.0  36.0
16  12.0      3.0       8.0      8.0  43.0
17  12.0      9.0       7.0      4.0  45.0
18  13.0      8.0       8.0      5.0  54.0
19  14.0      6.0       4.0      7.0  39.0
20  13.0      9.0       1.0      3.0  32.0
21  12.0      8.0       8.0      8.0  37.0
22  12.0      4.0       1.0      9.0  35.0
23  12.0      8.0       8.0      2.0  66.0
24  13.0      2.0       4.0      1.0  67.0
25  14.0      9.0       7.0      5.0  51.0
26  12.0      6.0       7.0      3.0  43.0
27  12.0      6.0       8.0      9.0  57.0
28  14.0      5.0       5.0      8.0  43.0
29  14.0      8.0       5.0      1.0  44.0
30  14.0      3.0       1.0      8.0  53.0
31  14.0      2.0       8.0      9.0  46.0
32  13.0      5.0       1.0      2.0  58.0
33  14.0      9.0       3.0      4.0  33.0
34  12.0      2.0       3.0      8.0  32.0
35  14.0      3.0       7.0      1.0  44.0
36  12.0      3.0       7.0      7.0  30.0
37  12.0      3.0       8.0      8.0  59.0
38  12.0      1.0       7.0      6.0  41.0
39  12.0      2.0       5.0      7.0  65.0
40  13.0      1.0       7.0      6.0  59.0

In [23]:
student_score.columns

Index(['Age', 'FirstCA', 'SecondCA', 'ThirdCA', 'Exam'], dtype='object')

In [24]:
student_score.rename({'Age': 'Student_age', 'Exam': 'Exam_score'}, axis=1,inplace=True)
student_score

Student_age  FirstCA  SecondCA  ThirdCA  Exam_score
1          13.0      6.0       9.0      6.0        43.0
2          14.0      1.0       1.0      2.0        63.0
3          12.0      8.0       7.0      3.0        50.0
4          12.0      5.0       6.0      3.0        40.0
5          14.0      5.0       3.0      5.0        62.0
6          12.0      8.0       8.0      2.0        55.0
7          13.0      8.0       1.0      7.0        51.0
8          12.0      8.0       7.0      2.0        69.0
9          14.0      1.0       2.0      9.0        62.0
10         12.0      9.0       4.0      9.0        49.0
11         14.0      8.0       4.0      7.0        55.0
12         14.0      6.0       2.0      4.0        44.0
13         12.0      5.0       9.0      2.0        43.0
14         13.0      5.0       1.0      4.0        50.0
15         12.0      3.0       1.0      5.0        36.0
16         12.0      3.0       8.0      8.0        43.0
17         12.0      9.0       7.0      4.0        45.0
18         13.0      8.0       8.0      5.0        54.0
19         14.0      6.0       4.0      7.0        39.0
20         13.0      9.0       1.0      3.0        32.0
21         12.0      8.0       8.0      8.0        37.0
22         12.0      4.0       1.0      9.0        35.0
23         12.0      8.0       8.0      2.0        66.0
24         13.0      2.0       4.0      1.0        67.0
25         14.0      9.0       7.0      5.0        51.0
26         12.0      6.0       7.0      3.0        43.0
27         12.0      6.0       8.0      9.0        57.0
28         14.0      5.0       5.0      8.0        43.0
29         14.0      8.0       5.0      1.0        44.0
30         14.0      3.0       1.0      8.0        53.0
31         14.0      2.0       8.0      9.0        46.0
32         13.0      5.0       1.0      2.0        58.0
33         14.0      9.0       3.0      4.0        33.0
34         12.0      2.0       3.0      8.0        32.0
35         14.0      3.0       7.0      1.0        44.0
36         12.0      3.0       7.0      7.0        30.0
37         12.0      3.0       8.0      8.0        59.0
38         12.0      1.0       7.0      6.0        41.0
39         12.0      2.0       5.0      7.0        65.0
40         13.0      1.0       7.0      6.0        59.0

In [25]:
student_score.rename({1: 'ONE',5:'FIVE'},axis=0,inplace=True)

In [26]:
student_score

Student_age  FirstCA  SecondCA  ThirdCA  Exam_score
ONE          13.0      6.0       9.0      6.0        43.0
2            14.0      1.0       1.0      2.0        63.0
3            12.0      8.0       7.0      3.0        50.0
4            12.0      5.0       6.0      3.0        40.0
FIVE         14.0      5.0       3.0      5.0        62.0
6            12.0      8.0       8.0      2.0        55.0
7            13.0      8.0       1.0      7.0        51.0
8            12.0      8.0       7.0      2.0        69.0
9            14.0      1.0       2.0      9.0        62.0
10           12.0      9.0       4.0      9.0        49.0
11           14.0      8.0       4.0      7.0        55.0
12           14.0      6.0       2.0      4.0        44.0
13           12.0      5.0       9.0      2.0        43.0
14           13.0      5.0       1.0      4.0        50.0
15           12.0      3.0       1.0      5.0        36.0
16           12.0      3.0       8.0      8.0        43.0
17           12.0      9.0       7.0      4.0        45.0
18           13.0      8.0       8.0      5.0        54.0
19           14.0      6.0       4.0      7.0        39.0
20           13.0      9.0       1.0      3.0        32.0
21           12.0      8.0       8.0      8.0        37.0
22           12.0      4.0       1.0      9.0        35.0
23           12.0      8.0       8.0      2.0        66.0
24           13.0      2.0       4.0      1.0        67.0
25           14.0      9.0       7.0      5.0        51.0
26           12.0      6.0       7.0      3.0        43.0
27           12.0      6.0       8.0      9.0        57.0
28           14.0      5.0       5.0      8.0        43.0
29           14.0      8.0       5.0      1.0        44.0
30           14.0      3.0       1.0      8.0        53.0
31           14.0      2.0       8.0      9.0        46.0
32           13.0      5.0       1.0      2.0        58.0
33           14.0      9.0       3.0      4.0        33.0
34           12.0      2.0       3.0      8.0        32.0
35           14.0      3.0       7.0      1.0        44.0
36           12.0      3.0       7.0      7.0        30.0
37           12.0      3.0       8.0      8.0        59.0
38           12.0      1.0       7.0      6.0        41.0
39           12.0      2.0       5.0      7.0        65.0
40           13.0      1.0       7.0      6.0        59.0

In [27]:
#student_score.replace({13:14},inplace=True)
#student_score['FirstCA'].replace({5:7},inplace=True)
#student_score.iloc[0,0] = 15

In [28]:
student_score['Student_age'].unique()

array([13., 14., 12.])

In [29]:
student_score['Student_age'].nunique()

3

In [34]:
student_score['Student_age'].value_counts()

12.0    19
14.0    13
13.0     8
Name: Student_age, dtype: int64

# assignment

In [3]:
months = ['Jan', 'Feb', 'Mar','Apr','May','Jun','Jul','Aug','Sept','Oct','Nov','Dec']
month_index = np.arange(1,31)
np.random.seed(10)
vals = np.random.randint(5000,500001,360)
vals = vals.reshape(30,12)

calender = pd.DataFrame(data = vals, index= month_index, columns = months)
calender

Jan     Feb     Mar     Apr     May     Jun     Jul     Aug    Sept  \
1   350353  241669  361879  448712   98553  110595   14372  408417  304741   
2   441218  145296  276433  191624  232818  361392  474494  250156  445691   
3    56595  465854  303328   12290  431558  430505  329808  226028   30550   
4   426965  479565  279445  405115  349473  186630  218645  172772  268510   
5   237076  241865  396711   36545  334508   54435   33178  209893  377813   
6   179840   25875  346660  413650  100448  340241   27543  221594  180568   
7   297235   72217   96566  202767  290387   30372  437929  199004  319106   
8   171533  351116  392143  372960   43744  211642  276724  340030  113834   
9    56855  319365  211982  333950  286173  265842  196126  129427  437752   
10  324550  334206  216129   29561  412119  318708  250992  139698  417397   
11   56757  363312  492385  380497  422445  478728   22298   85407  104843   
12  137434  231928  119190  475067  186327  192635  214681  230970  466877   
13  149618   44697   33064  422105  143985  249777   35810  197780   31098   
14  112516  172056  183304  255527  485613   42065  293704  391804  263428   
15  205288   49866  126923  361798  454631  337131  153562  105414  320126   
16  260024  162045  469861  289397  234835  449401  133627  426334  434477   
17  475699   54112  360040  227949  348113  452144  434413  233779  295539   
18  367824  415012  465031  403189  307622  411651  350049  475681  207958   
19   26517   57905   65321  155020  274015  194790  208292   46511  140515   
20  369694  236456   41611   88124  470868  240204  142374  297294  317302   
21  482321  171146  164681  117122  326521   61294  230907   16754  194069   
22  220261  224119  386268   12088  279366  285362  493098  379952  279478   
23  106462  194626  389453  147712   52122  137530  123256   43444   82185   
24  168346  388899  378331  475693  353615  417120   71663  460311  226239   
25  441018  134024  212404  281804  256150   93557  402429  289036  177934   
26  230648  275075  142188  495970    7854  445724  191249  361904  230936   
27   66649  199087  269753  443399  373609  490771  129110  195826   40663   
28   24258  307477  231664   17803  489556  497689   63269  152909  240186   
29  206975  469748  427598  205893   76912  484120   82822  196178  267788   
30  114257  471211   44544  388267  471032   43155   44465  190363  144561   

       Oct     Nov     Dec  
1   285048  113445  283385  
2   108952  320748  185463  
3    84537  217659  475070  
4    66214  265766  144353  
5   326942  208671  302566  
6   218622  403407  319712  
7   479976  413024  152500  
8   480496  315031  429792  
9   353940  446052  206184  
10  355521  304755   57018  
11  351909  176792  233692  
12   87841   76022  111416  
13  128166   45145  472345  
14   44334   40582  365982  
15  317909  185730  477427  
16  311955  157469   97767  
17  386274  421392  422493  
18  332121   26717   25414  
19  189423  304785  370831  
20  419072  226044  257234  
21   98643  337212  370189  
22  297768  432511  206054  
23  150727   38643   73301  
24  187150  393908   28561  
25   85938  239697   90548  
26   27859  254255  357570  
27  450973  488781  451896  
28  301499   66076  203661  
29  272158   56987  454109  
30   94041  126533   48402

In [26]:
mean_sales = (f'The average sales for Jan is {int(np.mean(calender["Jan"]))}, for May is {int(np.mean(calender["May"]))}, for Aug is {int(np.mean(calender["Aug"]))} and for Sept is {int(np.mean(calender["Sept"]))}')

print(mean_sales)

The average sales for Jan is 233777, for May is 293049, for Aug is 243559 and for Sept is 252549


In [15]:
calender.loc['max_sales'] = np.max(calender.iloc[0:30,0:12]) #retrieving a subset of rows and columns
calender

C:\Users\hp\anaconda3\lib\site-packages\numpy\core\fromnumeric.py:84: FutureWarning: In a future version, DataFrame.max(axis=None) will return a scalar max over the entire DataFrame. To retain the old behavior, use 'frame.max(axis=0)' or just 'frame.max()'
  return reduction(axis=axis, out=out, **passkwargs)


Jan     Feb     Mar     Apr     May     Jun     Jul     Aug  \
1          350353  241669  361879  448712   98553  110595   14372  408417   
2          441218  145296  276433  191624  232818  361392  474494  250156   
3           56595  465854  303328   12290  431558  430505  329808  226028   
4          426965  479565  279445  405115  349473  186630  218645  172772   
5          237076  241865  396711   36545  334508   54435   33178  209893   
6          179840   25875  346660  413650  100448  340241   27543  221594   
7          297235   72217   96566  202767  290387   30372  437929  199004   
8          171533  351116  392143  372960   43744  211642  276724  340030   
9           56855  319365  211982  333950  286173  265842  196126  129427   
10         324550  334206  216129   29561  412119  318708  250992  139698   
11          56757  363312  492385  380497  422445  478728   22298   85407   
12         137434  231928  119190  475067  186327  192635  214681  230970   
13         149618   44697   33064  422105  143985  249777   35810  197780   
14         112516  172056  183304  255527  485613   42065  293704  391804   
15         205288   49866  126923  361798  454631  337131  153562  105414   
16         260024  162045  469861  289397  234835  449401  133627  426334   
17         475699   54112  360040  227949  348113  452144  434413  233779   
18         367824  415012  465031  403189  307622  411651  350049  475681   
19          26517   57905   65321  155020  274015  194790  208292   46511   
20         369694  236456   41611   88124  470868  240204  142374  297294   
21         482321  171146  164681  117122  326521   61294  230907   16754   
22         220261  224119  386268   12088  279366  285362  493098  379952   
23         106462  194626  389453  147712   52122  137530  123256   43444   
24         168346  388899  378331  475693  353615  417120   71663  460311   
25         441018  134024  212404  281804  256150   93557  402429  289036   
26         230648  275075  142188  495970    7854  445724  191249  361904   
27          66649  199087  269753  443399  373609  490771  129110  195826   
28          24258  307477  231664   17803  489556  497689   63269  152909   
29         206975  469748  427598  205893   76912  484120   82822  196178   
30         114257  471211   44544  388267  471032   43155   44465  190363   
max_sales  482321  479565  492385  495970  489556  497689  493098  475681   

             Sept     Oct     Nov     Dec  
1          304741  285048  113445  283385  
2          445691  108952  320748  185463  
3           30550   84537  217659  475070  
4          268510   66214  265766  144353  
5          377813  326942  208671  302566  
6          180568  218622  403407  319712  
7          319106  479976  413024  152500  
8          113834  480496  315031  429792  
9          437752  353940  446052  206184  
10         417397  355521  304755   57018  
11         104843  351909  176792  233692  
12         466877   87841   76022  111416  
13          31098  128166   45145  472345  
14         263428   44334   40582  365982  
15         320126  317909  185730  477427  
16         434477  311955  157469   97767  
17         295539  386274  421392  422493  
18         207958  332121   26717   25414  
19         140515  189423  304785  370831  
20         317302  419072  226044  257234  
21         194069   98643  337212  370189  
22         279478  297768  432511  206054  
23          82185  150727   38643   73301  
24         226239  187150  393908   28561  
25         177934   85938  239697   90548  
26         230936   27859  254255  357570  
27          40663  450973  488781  451896  
28         240186  301499   66076  203661  
29         267788  272158   56987  454109  
30         144561   94041  126533   48402  
max_sales  466877  480496  488781  477427

In [20]:
feb_percentile = (f'The 20th percentile for Feb is {np.percentile(calender["Feb"],20)}, the 50th is {np.percentile(calender["Feb"],50)}, and the 75th is {np.percentile(calender["Feb"],75)}.')
jun_percentile = (f'The 20th percentile for Jun is {np.percentile(calender["Jun"],20)}, the 50th is {np.percentile(calender["Jun"],50)}, and the 75th is {np.percentile(calender["Jun"],75)}.')
dec_percentile = (f'The 20th percentile for Dec is {np.percentile(calender["Dec"],20)}, the 50th is {np.percentile(calender["Dec"],50)}, and the 75th is {np.percentile(calender["Dec"],75)}.')

print(feb_percentile)
print('')
print(jun_percentile)
print('')
print(dec_percentile)

The 20th percentile for Feb is 134024.0, the 50th is 236456.0, and the 75th is 357214.0.

The 20th percentile for Jun is 110595.0, the 50th is 285362.0, and the 75th is 438114.5.

The 20th percentile for Dec is 97767.0, the 50th is 257234.0, and the 75th is 396662.0.


In [30]:
total_salesJun = calender['Jun'].sum()
total_salesJul = calender['Jul'].sum()

print(f'The total sales in Jun was {total_salesJun}')
print(f'The total sales in Jul was {total_salesJul}')


if total_salesJun < total_salesJul:
    print(f'Difference between total sales in Jun and Jul is {total_salesJun - total_salesJul}, there were less sales in Jun compared to Jul')
elif total_salesJun > total_salesJul:
    print(f'Difference between total sales in Jun and Jul is {total_salesJun - total_salesJul}, there were more sales in Jun compared to Jul')
else:
    print(f'Difference between total sales in Jun and Jul is {total_salesJun - total_salesJul}, there were no differences in sales between Jun and Jul')
    
    

The total sales in Jun was 8812899
The total sales in Jul was 6573987
Difference between total sales in Jun and Jul is 2238912, there were more sales in Jun compared to Jul


In [31]:
calender.loc['total_sales'] = calender.iloc[0:30,0:12].sum() #retrieving a subset of rows and columns
calender

Jan      Feb      Mar      Apr      May      Jun      Jul  \
1             350353   241669   361879   448712    98553   110595    14372   
2             441218   145296   276433   191624   232818   361392   474494   
3              56595   465854   303328    12290   431558   430505   329808   
4             426965   479565   279445   405115   349473   186630   218645   
5             237076   241865   396711    36545   334508    54435    33178   
6             179840    25875   346660   413650   100448   340241    27543   
7             297235    72217    96566   202767   290387    30372   437929   
8             171533   351116   392143   372960    43744   211642   276724   
9              56855   319365   211982   333950   286173   265842   196126   
10            324550   334206   216129    29561   412119   318708   250992   
11             56757   363312   492385   380497   422445   478728    22298   
12            137434   231928   119190   475067   186327   192635   214681   
13            149618    44697    33064   422105   143985   249777    35810   
14            112516   172056   183304   255527   485613    42065   293704   
15            205288    49866   126923   361798   454631   337131   153562   
16            260024   162045   469861   289397   234835   449401   133627   
17            475699    54112   360040   227949   348113   452144   434413   
18            367824   415012   465031   403189   307622   411651   350049   
19             26517    57905    65321   155020   274015   194790   208292   
20            369694   236456    41611    88124   470868   240204   142374   
21            482321   171146   164681   117122   326521    61294   230907   
22            220261   224119   386268    12088   279366   285362   493098   
23            106462   194626   389453   147712    52122   137530   123256   
24            168346   388899   378331   475693   353615   417120    71663   
25            441018   134024   212404   281804   256150    93557   402429   
26            230648   275075   142188   495970     7854   445724   191249   
27             66649   199087   269753   443399   373609   490771   129110   
28             24258   307477   231664    17803   489556   497689    63269   
29            206975   469748   427598   205893    76912   484120    82822   
30            114257   471211    44544   388267   471032    43155    44465   
max_sales     482321   479565   492385   495970   489556   497689   493098   
total_sales  6764786  7299829  7884890  8091598  8594972  8315210  6080889   

                 Aug     Sept      Oct      Nov      Dec  
1             408417   304741   285048   113445   283385  
2             250156   445691   108952   320748   185463  
3             226028    30550    84537   217659   475070  
4             172772   268510    66214   265766   144353  
5             209893   377813   326942   208671   302566  
6             221594   180568   218622   403407   319712  
7             199004   319106   479976   413024   152500  
8             340030   113834   480496   315031   429792  
9             129427   437752   353940   446052   206184  
10            139698   417397   355521   304755    57018  
11             85407   104843   351909   176792   233692  
12            230970   466877    87841    76022   111416  
13            197780    31098   128166    45145   472345  
14            391804   263428    44334    40582   365982  
15            105414   320126   317909   185730   477427  
16            426334   434477   311955   157469    97767  
17            233779   295539   386274   421392   422493  
18            475681   207958   332121    26717    25414  
19             46511   140515   189423   304785   370831  
20            297294   317302   419072   226044   257234  
21             16754   194069    98643   337212   370189  
22            379952   279478   297768   432511   206054  
23             43444    82185   150727    38643    73301  
24            460311   226

# sales fluctuated between Jan and Jul with a high of 8,594,972 in May and a low of 6,080,889 in Jul. 

# there were steady sales between Aug and Dec as they ranged between 7,074,670 and 7,674,935

# in all sales 